node.js

```diff
@@ -48,7 +48,44 @@ module.exports = {
     'space-before-function-paren': off,
     'keyword-spacing': error,
     'eol-last': [ error, always ],
-    'brace-style': [ error, '1tbs', { allowSingleLine: true } ],
-    'curly': [ error, all ]
+    'brace-style': [ error, '1tbs', { allowSingleLine: false } ],
+    'curly': [ error, all ],
+    'block-spacing': [ error,  always ],
+    'indent': [ error, 2, { SwitchCase: 1 } ],
+    'no-tabs': error,
+    'spaced-comment': [ error, always ],
+    'array-callback-return': warn,
+    'eqeqeq': [ error, always ],
+    'no-else-return': [ warn, { allowElseIf: false } ],
+    'no-await-in-loop': warn,
+    'yoda': error,
+    'no-throw-literal': error,
+    'no-self-compare': error,
+    'no-return-await': warn,
```

Comment:
I may be wrong but I think in case of error you'll loose stacktrace entry if you don't use await. Somebody should check it, if it's true, then recommendation in this rule is plain wrong.

Reply:
I've just had a search around to double check, and I don't think you'll lose any stacktrace, I believe it's the equivalent of writing...

```js
// no-return-await
return somethingPromisey()
  .then(a => somethingElsePromisey(a))

// with return-await
return somethingPromisey()
  .then(a => {
    return somethingElsePromisey(a)
      .then(b => return b)
  })
 }
```

Nobody seems 100% sure, but everywhere I've seen suggests no-return-await is better

Reply:
Stack traces seem to be fine, so I was wrong about that.

My main argument against is out, but still I have reservations as:
1. it's weirdly asymmetric, easy async should be matched by await, it's just implementation detail leaking out which allows for this construct of avoiding await on return.
2. because of 1. some flow analysis may be impaired by it
3. it may encourage to write invalid code because if you don't have await statements in the body of function, you may omit `async` modifier, this can have bad consequences
4. performance argument for it is silly as it can be optimized out by v8 and/or babel easily if it's really equivalent